lib/code_fund_web/controllers/api/ad_serve_controller.ex

```diff
@@ -138,6 +124,65 @@ defmodule CodeFundWeb.API.AdServeController do
 
   defp details_render(payload, conn), do: render(conn, "details.json", payload: payload)
 
+  defp payload(
+         %Advertisement{
+           image_url: image_url,
+           body: body,
+           campaign_id: campaign_id,
+           headline: headline,
+           small_image_object: small_image_object,
+           large_image_object: large_image_object
+         },
+         conn,
+         impression_id
+       ) do
+    %{
+      image: image_url,
+      small_image_url: Framework.FileStorage.url(small_image_object),
+      large_image_url: Framework.FileStorage.url(large_image_object),
+      link: "https://#{conn.host}/c/#{impression_id}",
+      description: body,
+      pixel: "//#{conn.host}/p/#{impression_id}/pixel.png",
+      poweredByLink: "https://codefund.io?utm_content=#{campaign_id}",
+      headline: headline
+    }
+  end
+
+  defp create_impression_with_error(
+         conn,
+         property_id,
+         reason_message,
+         :no_possible_ads,
+         height,
+         width
+       ) do
+    params = %{
+      property_id: property_id,
+      error_code: AdService.ImpressionErrors.fetch_code(:no_possible_ads),
+      ip: conn.remote_ip |> Tuple.to_list() |> Enum.join("."),
+      browser_height: height,
+      browser_width: width
+    }
+
+    case property_id |> AdService.Query.ForDisplay.fallback_ad_by_property_id() do
+      %Advertisement{} = advertisement ->
+        {:ok, %Impression{id: impression_id}} =
+          params
+          |> Map.merge(%{campaign_id: advertisement.campaign_id, house_ad: true})
```

Comment:
Can we ensure that the revenue_amount and distribution_amounts here are set to zero?

Reply:
They are by default